### ISSUE TITLE
Possible inaccuracy in 'Defensive Programming'

### ISSUE THREAD

**Issue by dndln** [swcarpentry/python-novice-inflammation#329]
https://swcarpentry.github.io/python-novice-inflammation/08-defensive/

Section: Testing Assertions
```
def running(values):
    assert len(values) > 0
    result = [values[0]]
    for v in values[1:]:
        assert result[-1] >= 0
        result.append(result[-1] + v)
        assert result[-1] >= result[0]
    return result
```

> The second assertion checks that the first value in the list is positive. Input such as [-1,0,2,3] will make it fail.

The second assertion is actually checking if the last value of the 'result' list is positive on every iteration. On the first iteration it does check that the first value is positive, since result[0] == result[-1], but on subsequent iterations it does not.

**Comment by tbekolay** [comment]
Yes, you're quite right, a PR clarifying that would be great!

I don't recall when we added this exercise, but it strikes me as an odd one... why would we want to constrain the function to only accept monotonically increasing positive numbers? It's also not at all clear what would happen if we ran `running('abc')` with the 'defensive' code, as `'a' >= 0` is not obvious.

Perhaps the whole exercise should be replaced with something more straightforward?
